Replaces world.icon_size (and some magic numbers) with defines

tontyGH opened the pull request:
## About The Pull Request

All usages of world.icon_size in code have been replaced with new `ICONSIZE_X`, `ICONSIZE_Y` and `ICONSIZE_ALL` defines depending on context

Replaces some "32" magic numbers with the defines

A few bits of code have been modified to split up x/y math as well

## Why It's Good For The Game

Magic number bad, code more readable, code more flexible and I'm told there's an access cost to doing world.icon_size so minor performance gains

## Changelog

:cl: tonty
code: made some code relating to the world's icon size more readable
/:cl:

tontyGH committed:
this is the first one

tontyGH committed:
this is the first one

tontyGH committed:
Merge branch 'most-NEET-pr-ever' of https://github.com/tontyGH/fuckedupstation into most-NEET-pr-ever

tontyGH committed:
Caught some errors

tontyGH committed:
i thought it made more sense to be put in world.dm but oh well

Fikou reviewed (dismissed):
id prefer names ICON_SIZE, ICON_SIZE_X and ICON_SIZE_Y
i would actually set icon_size in world.dm to this new define
and i am not convinced about having iconsize_x and y to begin with, makes you have to think about which one to use everytime just in case we switch to something that isnt 1:1 ratio which seems incredibly improbable, i would like to know your thoughts on it

tontyGH committed:
ICONSIZE -> ICON_SIZE

tontyGH committed:
fuck

tontyGH committed:
Forgot a few magic numbers in get_turf_in_angle

tontyGH committed:
inLineOfSight is less hardcoded now, more readable
I figured this was relevant enough to what I was doing
And I'm sure there will be a coder sometime down the line who will thank me

tontyGH committed:
Merge branch 'master' of https://github.com/tontyGH/fuckedupstation into most-NEET-pr-ever

tontyGH committed:
spacemove

tontyGH committed:
found a few more...

Ghommie committed:
Merge branch 'master' into most-NEET-pr-ever
